src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterFactory.cs

```diff
@@ -33,6 +33,6 @@ internal JsonConverter GetConverterInternal(Type typeToConvert, JsonSerializerOp
         /// <returns>
         /// An instance of a <see cref="JsonConverter{T}"/> where T is compatible with <paramref name="typeToConvert"/>.
         /// </returns>
-        public abstract JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options);
+        public abstract JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions? options);
```

Comment:
@steveharter would know better, but I don't see why `options` should be nullable here.

Reply:
options was not used for all existing implementations

Reply:
> options was not used for all existing implementations

Even so, the contract is that the serializer will never pass null `options` and users who implement this can rely on it/not have to guard against null. In some cases, folks would want to call `options.GetConverter(...)` and that's when they would use the options. To me, the fact that options is not being used in the built-in converters doesn't indicate it can be null.

I discussed this offline as well. The `options` should not be null here.

```suggestion
        public abstract JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options);
```

Reply:
Sure will update, just explained why I make it nullable

Reply:
> just explained why I make it nullable

Is there anywhere we end up passing null options to this method?